code/datums/mutations/adaptation.dm

```diff
@@ -1,53 +1,58 @@
-/datum/mutation/human/temperature_adaptation
-	name = "Temperature Adaptation"
+/datum/mutation/human/adaptation
+	name = "Adaptation"
+	desc = "A strange mutation that adapts or something."
 	desc = "A strange mutation that renders the host immune to damage from extreme temperatures. Does not protect from vacuums."
 	quality = POSITIVE
 	difficulty = 16
-	text_gain_indication = "<span class='notice'>Your body feels warm!</span>"
-	instability = 25
-	conflicts = list(/datum/mutation/human/pressure_adaptation)
-
-/datum/mutation/human/temperature_adaptation/New(class_ = MUT_OTHER, timer, datum/mutation/human/copymut)
+	text_gain_indication = "<span class='notice'>Your body feels gay!</span>"
+	instability = 20
+	locked = TRUE // fake parent
+	conflicts = list(/datum/mutation/human/adaptation)
+	mutation_traits = list(TRAIT_WADDLING)
+	var/adapt_icon = "meow"
+
+/datum/mutation/human/adaptation/New(class_ = MUT_OTHER, timer, datum/mutation/human/copymut)
 	..()
 	if(!(type in visual_indicators))
-		visual_indicators[type] = list(mutable_appearance('icons/mob/effects/genetics.dmi', "fire", -MUTATIONS_LAYER))
+		visual_indicators[type] = list(mutable_appearance('icons/mob/effects/genetics.dmi', adapt_icon, -MUTATIONS_LAYER))
 
-/datum/mutation/human/temperature_adaptation/get_visual_indicator()
+/datum/mutation/human/adaptation/get_visual_indicator()
 	return visual_indicators[type][1]
 
-/datum/mutation/human/temperature_adaptation/on_acquiring(mob/living/carbon/human/owner)
-	if(..())
-		return
-	owner.add_traits(list(TRAIT_RESISTCOLD, TRAIT_RESISTHEAT), GENETIC_MUTATION)
-
-/datum/mutation/human/temperature_adaptation/on_losing(mob/living/carbon/human/owner)
-	if(..())
-		return
-	owner.remove_traits(list(TRAIT_RESISTCOLD, TRAIT_RESISTHEAT), GENETIC_MUTATION)
-
-/datum/mutation/human/pressure_adaptation
+/datum/mutation/human/adaptation/cold
+	name = "Cold Adaptation"
+	desc = "A strange mutation that renders the host immune to damage from low temperature environments. It also prevents the host from slipping on water or ice."
+	difficulty = 16
+	text_gain_indication = "<span class='notice'>Your body feels refreshingly cold.</span>"
+	instability = 20
+	mutation_traits = list(TRAIT_RESISTCOLD, TRAIT_NO_SLIP_WATER, TRAIT_NO_SLIP_ICE, TRAIT_NO_SLIP_SLIDE)
+	adapt_icon = "cold"
+	locked = FALSE
+
+/datum/mutation/human/adaptation/heat
+	name = "Heat Adaptation"
+	desc = "A strange mutation that renders the host immune to damage from high temperature environments. It also prevents the host from being ignited."
+	text_gain_indication = "<span class='notice'>Your body feels refreshingly warm.</span>"
+	instability = 20
+	mutation_traits = list(TRAIT_RESISTHEAT, TRAIT_NOFIRE)
+	adapt_icon = "fire"
+	locked = FALSE
```

Comment:
I think the ignition immunity is a bit too powerful.
This will essentially make people immune to the consequences of plasmafloods because their clothes won't burn off, etc.
I don't mind if you become immune to the damage from ignition, but your clothes and stuff should still burn off so that you can't get off scot free from walking through a fire.

Reply:
fire doesnt hurt you outright it just raises body temperature
so by default it 'removes' fire damage

im making it give ash storm immunity instead, that seems fun for miners